RFS/src/main/java/com/rfs/common/OpenSearchClient.java

```diff
@@ -243,27 +227,22 @@ public Optional<ObjectNode> getDocument(String indexName, String documentId) {
     }
 
     /*
-     * Update a document using optimistic locking.  Returns an Optional; if the document was updated, it
-     * will be the new value and empty otherwise.
+     * Update a document using optimistic locking with the versioning info in an original copy of the doc that is passed in.
+     * Returns an Optional; if the document was updated, it will be the new value and empty otherwise.
      */
-    public Optional<ObjectNode> updateDocument(String indexName, String documentId, ObjectNode body) {
-        Optional<ObjectNode> document = getDocument(indexName, documentId);
-        if (document.isEmpty()) {
-            throw new UpdateFailed("Document not found: " + indexName + "/" + documentId);
-        }
-
+    public Optional<ObjectNode> updateDocument(String indexName, String documentId, ObjectNode newBody, ObjectNode originalCopy) {
         String currentSeqNum;
         String currentPrimaryTerm;
         try {
-            currentSeqNum = document.get().get("_seq_no").asText();
-            currentPrimaryTerm = document.get().get("_primary_term").asText();
+            currentSeqNum = originalCopy.get("_seq_no").asText();
+            currentPrimaryTerm = originalCopy.get("_primary_term").asText();
         } catch (Exception e) {
             String errorMessage = "Could not update document: " + indexName + "/" + documentId;
             throw new RfsException(errorMessage, e);
         }
 
         ObjectNode upsertBody = new ObjectMapper().createObjectNode();
-        upsertBody.set("doc", body);
+        upsertBody.set("doc", newBody);
 
         String targetPath = indexName + "/_update/" + documentId + "?if_seq_no=" + currentSeqNum + "&if_primary_term=" + currentPrimaryTerm;
```

Comment:
Sequence number is on the shard of the index so that value will change if other operations happen on the shard unrelated to this document.  This code will be safer and more portable to future situations if you're able to do the optimistic locks without the sequence number